### PR TITLE
Analyze most of the source code with phpstan. Provide baseline.

### PR DESCRIPTION
**Pull request by tmotyl**
This patch makes phpstan analyze almost whole OpenMage source code.
To make the phpstan green despite not all issues are yet solved,
a baseline file is introduced, which contains "known issues".
Thanks to this approach we can make sure new changes
are not introducing new issues, and its easy to fix issues from baseline
one by one.

This patch also fixes some issues along the way:

1. deprecate _getHelper method from Mage_Adminhtml_Controller_Rss_Abstract
as it had incompatible interface with parent class, and was used only in one place.
2. Fix non existing class name Mage_Catalog_Model_Entity_Product_Collection
with correct one Mage_Catalog_Model_Resource_Product_Collection
3. Correct casing of the class name Zend_Pdf_Color_Rgb
4. few minor phpdocs changes
5. update macopedia/phpstan-magento1 to newest version which brings
performance improvements
6. add #[\ReturnTypeWillChange] in some places to avoid errors on php8.1

### Fixed Issues (if relevant)

1. Fixes OpenMage/magento-lts#365
2. Fixes OpenMage/magento-lts#1087

### Manual testing scenarios (*)

1. ...
3. ...

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

**Comment by tmotyl** [comment]
interestingly, here on github phpstan finds more issues then running locally. I'm investigating

**Comment by tmotyl** [comment]
mystery solved, its due to php version. locally I run it with 7.4 and github runs 8.1

**Comment by tmotyl** [comment]
looks good now

**Comment by sreichel** [comment]
I'd keep _getHelper() method. It could be used in 3rd party code.

Maybe mark it as deprecated and ignore it from sniffs?

**Comment by tmotyl** [comment]
> I'd keep _getHelper() method. It could be used in 3rd party code.
> 
> Maybe mark it as deprecated and ignore it from sniffs?

This issue cant be ignored afaik, only whole file can be skipped.
Alternatively i can keep the method but remove the param and make it always return rss helper? This would make it less breaking

**Comment by tmotyl** [comment]
@sreichel I've excluded the RSS files from analysing and deprecated the method. We might want to remove this method in v20 branch. but this will be a different patch.

**Comment by tmotyl** [comment]
FYI, once we have this one merged, I have a bunch of patches ready on top of it, which lowers down number of issues and rise the phpstan level to 1.
A little teaser:
image

**Review by sreichel — APPROVED** [review]

**Comment by sreichel** [comment]
> FYI, once we have this one merged, I have a bunch of patches ready on top of it,

NICE! Thanks for your effort. :heart: 

Not ready for PR, but i already tried to bring it to level 2,3 or 4 as where it is possible :)

File: app/code/core/Mage/Catalog/Model/Product/Attribute/Api.php [hunk 1]
```diff
@@ -217,10 +217,10 @@ public function update($attribute, $data)
             $model->save();
             // clear translation cache because attribute labels are stored in translation
             Mage::app()->cleanCache(array(Mage_Core_Model_Translate::CACHE_TAG));
-            return true;
         } catch (Exception $e) {
             $this->_fault('unable_to_save', $e->getMessage());
         }
+        return true;
```
Comment (Flyingmana) [inline app/code/core/Mage/Catalog/Model/Product/Attribute/Api.php:223]: thats a very strong behavior change to now return true even in case of an error?

Comment (tmotyl) [reply app/code/core/Mage/Catalog/Model/Product/Attribute/Api.php:223]: Is it? `$this->_fault('unable_to_save', $e->getMessage());` throws an exception internally, so return is not reached.

Comment (tmotyl) [reply app/code/core/Mage/Catalog/Model/Product/Attribute/Api.php:223]: see https://github.com/OpenMage/magento-lts/blob/d19f0e13d3d2fcbcc2d8fa6073568d5f95442470/app/code/core/Mage/Api/Model/Resource/Abstract.php#L97

Comment (sreichel) [reply app/code/core/Mage/Catalog/Model/Product/Attribute/Api.php:223]: @Flyingmana agree to _return is not reached_.

**Review by Flyingmana — APPROVED** [review]

**Pull request merged**